### PR TITLE
Add support for Swift conditional copy

### PR DESCRIPTION
**Pull request by andrewgaul**

### CODE REVIEW

File: apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/features/ObjectApi.java [hunk 1]
```diff
@@ -323,6 +351,7 @@ void copy(@PathParam("destinationObject") String destinationObject,
     *
     * @throws KeyNotFoundException if the source or destination container do not exist.
     */
+   // TODO: CopyOptions
```
Comment (andrewgaul) [inline apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/features/ObjectApi.java]: Need to update these two APIs as above.

**Comment by andrewgaul** [comment]
@zack-shoylev Please review.

**Comment by zack-shoylev** [comment]
Looks good, what about the TODOs?

File: apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/features/ObjectApi.java (hunk 1 above)
Comment (andrewgaul) [inline apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/features/ObjectApi.java]: Done.

**Pull request merged**